File: backend/src/main/java/com/bakery/service/CakeService.java
package com.bakery.service;

import com.bakery.model.Cake;
import com.bakery.repository.CakeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class CakeService {

    @Autowired
    private CakeRepository cakeRepository;

    public Cake addCake(Cake cake) {
        return cakeRepository.save(cake);
    }

    public List<Cake> getAllCakes() {
        return cakeRepository.findAll();
    }

    public Optional<Cake> getCakeById(Long id) {
        return cakeRepository.findById(id);
    }

    public List<Cake> searchCakes(String name) {
        return cakeRepository.findByNameContainingIgnoreCase(name);
    }

    public List<Cake> getCakesByType(String type) {
        return cakeRepository.findByCakeType(type);
    }

    public Cake updateCake(Long id, Cake updatedCake) {
        return cakeRepository.findById(id).map(cake -> {
            cake.setName(updatedCake.getName());
            cake.setDescription(updatedCake.getDescription());
            cake.setPrice(updatedCake.getPrice());
            cake.setSize(updatedCake.getSize());
            cake.setFlavor(updatedCake.getFlavor());
            return cakeRepository.save(cake);
        }).orElseThrow(() -> new RuntimeException("Cake not found with id: " + id));
    }

    public void deleteCake(Long id) {
        cakeRepository.deleteById(id);
    }
}

File: frontend/src/services/cakeService.js
import api from './api'

export const getAllCakes = () => api.get('/cakes')
export const getCakeById = (id) => api.get(`/cakes/${id}`)
export const addCake = (data) => api.post('/cakes', data)
export const addBirthdayCake = (data) => api.post('/cakes/birthday', data)
export const addWeddingCake = (data) => api.post('/cakes/wedding', data)
export const updateCake = (id, data) => api.put(`/cakes/${id}`, data)
export const deleteCake = (id) => api.delete(`/cakes/${id}`)
export const searchCakes = (name) => api.get(`/cakes/search?name=${name}`)
